### PR TITLE
(APPINT-34009) fix the tCosmosDB and tNeo4j dependency issues when bu…

### PR DESCRIPTION
**Pull request by xldai**
…ilding microservice

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java [hunk 1]
```diff
@@ -56,15 +56,15 @@
...
+                    .map(d -> Optional.ofNullable(d.getMavenUri())
+                            .map(uri -> uri.substring(uri.indexOf("/") + 1).replaceAll("/", "-"))
                             .orElseGet(() -> d.getModuleName()))
                     .filter(jar -> TACOKIT_JARS_BLACKLIST.stream().noneMatch(tck -> jar.contains(tck)))
                     .collect(Collectors.toSet());
             if (ComponentModel.class.isInstance(component)) {
                 // we keep plugin in original classloader classpath (intent for microservices spring boot)
                 final String pluginPrefix = ((ComponentModel) component).getId().getPlugin() + "-";
-                tckDeps.addAll(deps.stream().filter(dep -> !dep.startsWith(pluginPrefix)).collect(Collectors.toSet()));
+                tckDeps.addAll(deps.stream().filter(dep -> !dep.matches(pluginPrefix + "[0-9]+")).collect(Collectors.toSet()));
```
Comment (undx) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java:67]: I think that the correct regexp should be `"[0-9]+.*"`

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java [hunk 2]
```diff
@@ -56,15 +56,15 @@
         components.forEach(component -> {
             final Set<String> deps = component.getModulesNeeded()
                     .stream()
-                    .map(d -> Optional.ofNullable(d.getModuleLocaion())
-                            .map(loc -> loc.substring(loc.lastIndexOf("/") + 1))
+                    .map(d -> Optional.ofNullable(d.getMavenUri())
+                            .map(uri -> uri.substring(uri.indexOf("/") + 1).replaceAll("/", "-"))
```
Comment (undx) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java:60]: two issues here with using mavenUri: 
- all jars filenames will be incorrects. ie: the replaceall will lead smtg like `commons-logging-1.2-jar` (no extension).
- uri may contains a studio extra version like `6.0.0` or `6.0.0-SNAPSHOT`, so the the filename will be stmg like `commons-logging-1.2-6.0.0-jar`

A possible solution (to be tested), is to split mavenUri w/  `uri.split("/")` that will give an array like this: 
```
0 = "mvn:commons-logging"
1 = "commons-logging"
2 = "1.2"
3 = "jar"
``` 
You will discard `0` and concatenate `1` + `-` + `2` + ` .` + `3`
But, we still have the `6.0.0(-SNAPSHOT)*` issue! So, a check on version (`2`) may be needed.
In those cases, most of the time, the real version is stored in artifact name (`1`). 
I found an example w/ commons-codec :
 ```
0 = "mvn:org.talend.libraries"
1 = "commons-codec-1.15"
2 = "6.0.0"
3 = "jar"
```

**Review by undx — COMMENTED** [review]
Hi. some comments in PR.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java [hunk 3]
```diff
@@ -56,15 +56,15 @@
         components.forEach(component -> {
             final Set<String> deps = component.getModulesNeeded()
                     .stream()
-                    .map(d -> Optional.ofNullable(d.getModuleLocaion())
-                            .map(loc -> loc.substring(loc.lastIndexOf("/") + 1))
+                    .map(d -> Optional.ofNullable(d.getMavenUri())
+                            .map(uri -> uri.split("/")[1] + "-" + uri.split("/")[2] + ".jar")
```
Comment (undx) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java:60]: I think that the artifacts version containing `6.0.0` may lead,in some cases, to an issue (duplicated jars for instance) but not totally sure...

Comment (xldai) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/service/TaCoKitDependencyService.java:60]: then how we proceed?
I'm triggering a temp build on this branch.

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**